### PR TITLE
Implemented sign out function

### LINKED ISSUE #52: Give users the ability to sign out

### PR DESCRIPTION
**Pull request by vocatwk**
I created a signout servlet that just nulls out the user attribute. Let me know if you guys think there is a better way to go about this instead of creating an entire servlet.
Also it sort of pushed the search bar down. @lealgamb Let me know if you liked it better where it was 😂

I also edited navbar.jsp cause the link to user's profile page was broken and I edited PresistentDataStore because I had user entities that were created before isAdmin was implented and they were giving me a nullPointerexception

screen shot 2018-06-19 at 10 02 51 pm

resolves #52

### CODE REVIEW

File: src/main/java/codeu/controller/SignoutServlet.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the signout link. */
+public class SignoutServlet extends HttpServlet {
+
```
Comment (yebrahim) [inline src/main/java/codeu/controller/SignoutServlet.java]: nit: extra new line.

File: src/main/java/codeu/controller/SignoutServlet.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  
+  /**
+   * This function fires when a user navigates clicks on the signout link. It nulls out the
```
Comment (yebrahim) [inline src/main/java/codeu/controller/SignoutServlet.java]: typo: "navigates clicks"

File: src/main/java/codeu/controller/SignoutServlet.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package codeu.controller;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the signout link. */
+public class SignoutServlet extends HttpServlet {
```
Comment (yebrahim) [inline src/main/java/codeu/controller/SignoutServlet.java]: We're using `LoginServlet`, perhaps this should be `LogoutServlet`?

File: src/main/webapp/WEB-INF/view/navbar.jsp [hunk 4]
```diff
@@ -9,15 +9,16 @@
...
   <% if(navBarUsername != null){ %>
-    <a href="/profile/<%=navBarUser %>">
+    <a href="/profile/<%=navBarUsername %>">
       Hello <%= navBarUsername %>!</a>
   <% } else{ %>
     <a href="/login">Login</a>
   <% } %>
   <a href="/about.jsp">About</a>
   <a href="/activityfeed">Activity Feed</a>
+  <a href="/signout" style="float: right"> Sign out </a>
```
Comment (yebrahim) [inline src/main/webapp/WEB-INF/view/navbar.jsp]: If you'll take my rename suggestion above, then this should be `/logout` to match. :)

**Comment by vocatwk** [comment]
Thanks! I think I can do it in this PR.

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 5]
```diff
@@ -70,9 +70,9 @@ public PersistentDataStore() {
         Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
         User user = new User(uuid, userName, passwordHash, creationTime);
         String aboutMe = (String) entity.getProperty("about_me");
-        boolean isAdmin = (boolean) entity.getProperty("is_admin");
+        Boolean isAdmin = (Boolean) entity.getProperty("is_admin");
         if(aboutMe != null) user.setAboutMe(aboutMe);
-        if(isAdmin == true) user.setIsAdmin(isAdmin);
+        if(isAdmin != null && isAdmin == true) user.setIsAdmin(isAdmin);
```
Comment (lealgamb) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: Shouldn't we set is admin whether it's T/F?

**Comment by lealgamb** [comment]
LGTM! And we can probably fix the search bar location by adjusting the margins in the nav bar. 👍

**Comment by yebrahim** [comment]
I wouldn’t worry too much about the looks, we might want to invest in some frontend work anyway to give the app our team’s touch. :) 🐠🐟🐡🐋🐳🐬

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java (hunk 5 above)
Comment (vocatwk) [reply src/main/java/codeu/model/store/persistence/PersistentDataStore.java]: isAdmin is set to false initially in the constructor.

**Pull request merged**